server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepository.java

```diff
@@ -90,7 +90,8 @@ public Flux<UploadMetaData> listUploads(Username user) {
 
     public Mono<Void> purge() {
         Instant sevenDaysAgo = clock.instant().minus(EXPIRE_DURATION);
-        return Flux.from(uploadDAO.listByUploadDateLessThan(sevenDaysAgo))
+        return Flux.from(uploadDAO.all())
+            .filter(upload -> upload.getUploadDate().isBefore(sevenDaysAgo))
             .concatMap(upload -> Mono.from(blobStore.delete(UPLOAD_BUCKET, upload.getBlobId()))
                 .then(uploadDAO.delete(upload.getUser(), upload.getId())))
```

Comment:
Q: Should we use flatMap to fasten the deletion?

Reply:
A: likely